src/query/functions/temporal/base.go

```diff
@@ -49,15 +49,27 @@ type baseOp struct {
 // skipping lint check for a single operator type since we will be adding more
 // nolint : unparam
 func newBaseOp(args []interface{}, operatorType string, processorFn MakeProcessor) (baseOp, error) {
-	if operatorType != HoltWintersType && operatorType != PredictLinearType {
+	if operatorType != HoltWintersType && operatorType != PredictLinearType && operatorType != QuantileType {
 		if len(args) != 1 {
 			return emptyOp, fmt.Errorf("invalid number of args for %s: %d", operatorType, len(args))
 		}
 	}
 
-	duration, ok := args[0].(time.Duration)
+	var (
+		durArg   interface{}
+		duration time.Duration
+		ok       bool
+	)
+
+	if operatorType != QuantileType {
+		durArg = args[0]
+	} else {
+		durArg = args[1]
```

Comment:
I know you check outside of this function, but is it worth checking len args in this func just in case? There's some check above for len(args) != 1, but does that cover everything?

Reply:
Eh I think it's fine (we do the same thing in the Holt-Winters function) + newBaseOp() is a private function